Doc/reference/compound_stmts.rst

```diff
@@ -510,6 +511,620 @@ the items are surrounded by parentheses. For example::
+|                          |                   |                                                          |
+|                          |                   | 4. test ``p2`` against ``subject_value[1]``              |
+|                          |                   |                                                          |
+|                          |                   | 5. repeat step 4. for subsequent patterns and indexes    |
++--------------------------+-------------------+----------------------------------------------------------+
+| :ref:`mapping_patterns`  | ``{p1: p2, ...}`` | 1. check ``isinstance(subject, collections.abc.Mapping)``|
+|                          |                   |                                                          |
+|                          |                   | 2. test ``p1 in subject``                                |
+|                          |                   |                                                          |
+|                          |                   | 3. test ``p2`` matches ``subject[p1]``                   |
+|                          |                   |                                                          |
+|                          |                   | 4. repeat steps 2. and 3. for subsequent key and value   |
+|                          |                   |    patterns.                                             |
++--------------------------+-------------------+----------------------------------------------------------+
+| :ref:`class_patterns`    | ``K(p1, name=p2)``| 1. check ``isinstance(K, type)``                         |
+|                          |                   |                                                          |
+|                          |                   | 2. check ``isinstance(subject, K)``                      |
+|                          |                   |                                                          |
+|                          |                   | 3. check ``hasattr(subject, name)``                      |
+|                          |                   |                                                          |
+|                          |                   | 4. test ``p2`` matches ``subject.name``                  |
+|                          |                   |                                                          |
+|                          |                   | 5. convert ``p1`` to a keyword pattern using             |
+|                          |                   |    ``K.__match_args__`` and repeat steps 2. and 3.       |
++--------------------------+-------------------+----------------------------------------------------------+
+
+Note that this table purely for illustration purposes and **may not** reflect
+the underlying implementation.  Furthermore, they do not cover all valid forms.
+Read the pattern's respective sections to learn more.
```

Comment:
I think this table is nice for a quick presentation, but given that we're adding the whole spec below is somewhat redundant, I'm still undecided if we should leave it here or not